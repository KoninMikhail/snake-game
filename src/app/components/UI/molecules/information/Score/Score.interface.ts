import { TextAlign } from '@/types/typography';

export interface IScoreProps {
    align?: TextAlign;
}
