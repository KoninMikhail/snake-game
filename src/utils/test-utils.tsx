import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
