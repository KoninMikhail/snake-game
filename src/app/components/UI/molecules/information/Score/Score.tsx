import { HeavyCaption } from '@ui/atoms/typography/caption/HeavyCaption/HeavyCaption';
import padCharactersLeft from '@utils/padCharactersLeft/padCharactersLeft';
import { IScoreProps } from '@ui/molecules/information/Score/Score.interface';
import useAppSelector from '@hooks/useAppSelector/useAppSelector';
import { gameScoreSelector } from '@store/slices/game.slice';

/**
 * @name Score
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use it for display count engine score;
 *
 * @param align - text alignment
 */
export const Score = ({ align = 'left' }: IScoreProps) => {
    const score = useAppSelector(gameScoreSelector);
    const outputScore = padCharactersLeft(score, 6);
    return <HeavyCaption align={align}>{outputScore}</HeavyCaption>;
};
