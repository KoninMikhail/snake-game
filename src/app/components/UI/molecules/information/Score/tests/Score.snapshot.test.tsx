import {Score} from '../Score';
import {GameData} from '@store/slices/game.slice';
import {renderWithProviders} from '@utils/test-utils';

/**
 * Tests
 */
describe('Score', () => {
    it('equal snapshot', () => {
        const mockScore = 30;

        const initialState = {
            preloadedState: {game: {score: mockScore} as GameData},
        };
        const {container} = renderWithProviders(<Score/>, initialState);

        expect(container).toMatchSnapshot();
    });
});
