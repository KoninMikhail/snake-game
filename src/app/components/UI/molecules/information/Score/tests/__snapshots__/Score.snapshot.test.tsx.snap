// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score equal snapshot 1`] = `
.c0 {
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 2.8em;
  font-weight: normal;
  text-align: left;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.5em;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 3.4em;
  }
}

<div>
  <span
    class="c0"
  >
    000030
  </span>
</div>
`;
